### PR TITLE
Adapt ScmExtractor to support push events

### PR DESCRIPTION
**Pull request by danidoni**
Details on the webhook payloads:
- [GitHub push](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push)
- [GitLab push](https://docs.gitlab.com/ee/user/project/integrations/webhooks.html#push-events)

### CODE REVIEW

**Comment by dmarcoux** [comment]
[Adapt workflow steps for push events on branches](https://trello.com/c/KUFzNTDw/1749-adapt-workflow-steps-for-push-events-on-branches)

File: src/api/app/services/trigger_controller_service/scm_extractor.rb [hunk 1]
```diff
@@ -21,36 +21,60 @@ def call
...
+        payload.merge!({ commit_sha: @payload.dig(:pull_request, :head, :sha),
+                         pr_number: @payload[:number],
+                         source_branch: @payload.dig(:pull_request, :head, :ref),
+                         target_branch: @payload.dig(:pull_request, :base, :ref),
+                         action: @payload[:action], # TODO: Names may differ, maybe we need to find our own naming (defer to service?)
+                         source_repository_full_name: @payload.dig(:pull_request, :head, :repo, :full_name),
+                         target_repository_full_name: @payload.dig(:pull_request, :base, :repo, :full_name) })
+      when 'push'
+        payload.merge!({ commit_sha: @payload[:after],
+                         branch: @payload[:ref].split('/').last,
```
Comment (hennevogel) [inline src/api/app/services/trigger_controller_service/scm_extractor.rb:41]: branch names can include `/`

Comment (dmarcoux) [reply src/api/app/services/trigger_controller_service/scm_extractor.rb:41]: Yep, I already have code for this, it's not pushed yet though. I'm working on the specs...

File: src/api/app/services/trigger_controller_service/scm_extractor.rb [hunk 2]
```diff
@@ -21,36 +21,62 @@ def call
...
+                         source_repository_full_name: @payload.dig(:pull_request, :head, :repo, :full_name),
+                         target_repository_full_name: @payload.dig(:pull_request, :base, :repo, :full_name) })
+      when 'push'
+        payload.merge!({ commit_sha: @payload[:after],
+                         # We need this for Workflows::YAMLDownloader#download_url
+                         target_branch: @payload[:ref].sub('refs/heads/', ''),
+                         # We need this for Workflow::Step#branch_request_content_github
+                         source_repository_full_name: @payload.dig(:repository, :full_name),
+                         # We need this for SCMStatusReporter#call
+                         target_repository_full_name: @payload.dig(:repository, :full_name) })
```
Comment (rubhanazeem) [inline src/api/app/services/trigger_controller_service/scm_extractor.rb:46]: Is this correct? `source_repository_full_name` and `target_repository_full_name` are pointing towards same element?

Comment (dmarcoux) [reply src/api/app/services/trigger_controller_service/scm_extractor.rb:46]: Yes, see the comments above each line.

**Review by rubhanazeem — APPROVED** [review]

**Pull request merged**